src/AbstractHelper.php

```diff
@@ -57,12 +62,57 @@ public function getModuleCategory(): string
         return __('Custom Widgets', 'fabric');
     }
 
+    /**
+     * @param string|string[] $templates Possible template files.
+     * @return array
+     */
+    public function filter_templates($templates)
+    {
+        $paths = apply_filters('sage/filter_templates/paths', [
+            'views',
+            'resources/views'
+        ]);
+        $paths_pattern = '#^(' . implode('|', $paths) . ')/#';
+
+        return collect($templates)
+            ->map(function ($template) use ($paths_pattern) {
+                /** Remove .blade.php/.blade/.php from template names */
+                $template = preg_replace('#\.(blade\.?)?(php)?$#', '', ltrim($template));
+
+                /** Remove partial $paths from the beginning of template names */
+                if (strpos($template, '/')) {
+                    $template = preg_replace($paths_pattern, '', $template);
+                }
+
+                return $template;
+            })
+            ->flatMap(function ($template) use ($paths) {
+                return collect($paths)
+                    ->flatMap(function ($path) use ($template) {
+                        return [
+                            "{$path}/{$template}.blade.php",
+                            "{$path}/{$template}.php",
+                        ];
+                    })
+                    ->concat([
+                        "{$template}.blade.php",
+                        "{$template}.php",
+                    ]);
+            })
+            ->filter()
+            ->unique()
+            ->all();
+    }
+
     /**
      * @param string|string[] $templates Relative path to possible template files.
      *
      * @return string Location of the template
      */
-    abstract public function locateTemplate($templates): string;
+    public function locateTemplate($templates): string
```

Comment:
Why no longer abstract?

Reply:
@codepuncher, this is a redundant function, I have removed it.